xcb-types.el

```diff
@@ -778,12 +777,11 @@ and the second the consumed length."
                    (setq slot-type (cl--slot-descriptor-type slot))
                    (throw 'break nil))))
              (unless (eq slot-type 'xcb:-ignore)
-               (setq tmp (xcb:-unmarshal-field obj slot-type data offset
+               (setq tmp (xcb:-unmarshal-field obj slot-type data (+ offset count)
```

Comment:
I changed the code slightly here, using (+ offset count) instead of modifying offset, for consistency with another branch of the case expression.